### PR TITLE
scrivito: Update to Scrivito 1.21

### PR DESCRIPTION
**Pull request by eric-the-viking**
Please fill in this template.

- [X] Use a meaningful title for the pull request. Include the name of the package modified.
- [X] Test the change in your own code. (Compile and run.)
- [X] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [X] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [X] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [X] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If changing an existing definition:
- [X] Provide a URL to documentation or source code which provides context for the suggested changes: https://www.scrivito.com/scrivito-js-1-21-0-released-featuring-the-changes-sidebar-d08dc7db42e29336
- [X] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

### CODE REVIEW

**Comment by juliankrieger** [comment]
@DanielRosenwasser as the maintainer for these Definitions, I approve this PR

**Review by juliankrieger — APPROVED** [review]

**Comment by juliankrieger** [comment]
Approve again

File: types/scrivito/index.d.ts [hunk 1]
```diff
@@ -101,71 +102,73 @@ interface BackgroundImageTagProps {
...
+export class WidgetTag extends React.Component<WidgetTagProps, any> { }
 
 /**
  * Config definitions
  */
 
 interface ConfigOptions {
     tenant: string;
-    homepage: () => Obj;
+    homepage?: () => Obj | null;
```
Comment (amcasey) [inline types/scrivito/index.d.ts:161]: Just to confirm, you wanted `() => (Obj | null)` and not `(() => Obj) | null`, right?

File: types/scrivito/index.d.ts [hunk 2]
```diff
@@ -464,7 +476,7 @@ export class Team {
 export function canWrite(): boolean;
 export function configure(options: ConfigOptions): void;
 export function configureContentBrowser(options: any): void;
-export function connect(component: React.ComponentClass | React.FunctionComponent): void;
+export function connect<C>(component: C): C;
```
Comment (amcasey) [inline types/scrivito/index.d.ts:479]: Was there supposed to be a constraint on `C`?

**Review by amcasey — APPROVED** [review]
LGTM, but I had a couple questions.

File: types/scrivito/index.d.ts (hunk 1 above)
Comment (juliankrieger) [reply types/scrivito/index.d.ts:161]: Yea, sadly the function is entirely optional but may return "null", too.

File: types/scrivito/index.d.ts (hunk 2 above)
Comment (juliankrieger) [reply types/scrivito/index.d.ts:479]: Yea, on React Components. Thanks, I missed this

Comment (eric-the-viking) [reply types/scrivito/index.d.ts]: Hi @amcasey
thanks for your feedback. We added the constraints in ~f142068.
`export function connect<C extends React.ComponentType<any>>(component: C): C;`

**Review by juliankrieger — APPROVED** [review]

**Review by amcasey — APPROVED** [review]

**Comment by eric-the-viking** [comment]
Ready to merge

**Pull request merged**